### PR TITLE
PHP 8 support

### PR DESCRIPTION
**Pull request by bennothommo**
This PR brings PHP 8 support to October Rain.

Main PR: https://github.com/octobercms/october/pull/5357

Big thanks to @tolkamsoft for their assistance with this so far.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -66,9 +66,15 @@
...
         "php-parallel-lint/php-parallel-lint": "^1.0",
-        "meyfa/phpunit-assert-gd": "^2.0.0",
-        "dms/phpunit-arraysubset-asserts": "^0.1.0"
+        "meyfa/phpunit-assert-gd": "^2.0.0|dev-master",
+        "dms/phpunit-arraysubset-asserts": "^0.1.0|^0.2.1"
     },
+    "repositories": [
+        {
+            "type": "vcs",
+            "url": "https://github.com/bennothommo/phpunit-assert-gd"
```
Comment (LukeTowers) [inline composer.json]: If we have to retain this into release I would prefer it be under octoberrain

Comment (bennothommo) [reply composer.json]: Sure thing. Hopefully it gets merged though.

**Comment by Denoder** [comment]
Just putting it out here that i've tested this and it's working on my end. On 7.3, 7.4 and 8.0

File: src/Foundation/Maker.php [hunk 2]
```diff
@@ -119,8 +119,9 @@ protected function getDependencies(array $parameters, array $primitives = [])
     {
         $dependencies = [];
 
+        /** @var ReflectionParameter $parameter */
         foreach ($parameters as $parameter) {
-            $dependency = $parameter->getClass();
+            $dependency = $parameter->getType();
```
Comment (LukeTowers) [inline src/Foundation/Maker.php:124]: Did getType() exist in PHP 7.2?

Comment (Denoder) [reply src/Foundation/Maker.php:124]: yes it is (available since 7.0): https://php.watch/versions/8.0/deprecated-reflectionparameter-methods

Comment (LukeTowers) [reply src/Foundation/Maker.php:124]: Excellent. Thanks @Teranode!

File: src/Foundation/Exception/Handler.php [hunk 3]
```diff
@@ -2,13 +2,15 @@
...
 use Event;
+use ReflectionClass;
 use Response;
 use Illuminate\Foundation\Exceptions\Handler as ExceptionHandler;
 use Symfony\Component\HttpKernel\Exception\HttpExceptionInterface;
 use October\Rain\Exception\AjaxException;
 use ReflectionFunction;
 use Exception;
 use Closure;
+use Throwable;
```
Comment (LukeTowers) [inline src/Foundation/Exception/Handler.php:8]: @bennothommo can you octoberify this use block?

Comment (bennothommo) [reply src/Foundation/Exception/Handler.php:8]: @LukeTowers done :)

**Comment by LukeTowers** [comment]
LGTM @bennothommo, you can merge if you're happy with it!

**Pull request merged**